### PR TITLE
Group ALTER TABLE related things together

### PR DESCRIPTION
**Pull request by dveeden**
### What is changed, added or deleted? (Required)

Things like `DROP COLUMN` are not statements, they are *part* of the `ALTER TABLE` statement. Let's group these together.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [ ] v8.1 (TiDB 8.1 versions)
- [ ] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by dveeden** [comment]
Is this the right way or could we get rid of "Overview" and add the link to "ALTER TABLE"?

Do we want to do something similar with "ADMIN"?

Another option: Change the names, e.g. change `ADD INDEX` to `ALTER TABLE ... ADD INDEX` etc.

File: TOC.md [hunk 1]
```diff
@@ -714,8 +714,7 @@
       - [Expression Syntax](/expression-syntax.md)
       - [Comment Syntax](/comment-syntax.md)
     - SQL Statements
-      - [`ADD COLUMN`](/sql-statements/sql-statement-add-column.md)
-      - [`ADD INDEX`](/sql-statements/sql-statement-add-index.md)
+
```
Comment (hfxsd) [inline TOC.md:717]:
```suggestion
```

File: TOC.md [hunk 2]
```diff
@@ -728,22 +727,29 @@
...
-      - [`ALTER INDEX`](/sql-statements/sql-statement-alter-index.md)
       - [`ALTER INSTANCE`](/sql-statements/sql-statement-alter-instance.md)
       - [`ALTER PLACEMENT POLICY`](/sql-statements/sql-statement-alter-placement-policy.md)
       - [`ALTER RANGE`](/sql-statements/sql-statement-alter-range.md)
       - [`ALTER RESOURCE GROUP`](/sql-statements/sql-statement-alter-resource-group.md)
       - [`ALTER SEQUENCE`](/sql-statements/sql-statement-alter-sequence.md)
-      - [`ALTER TABLE`](/sql-statements/sql-statement-alter-table.md)
-      - [`ALTER TABLE COMPACT`](/sql-statements/sql-statement-alter-table-compact.md)
+      - `ALTER TABLE`
+        - [`Overview`](/sql-statements/sql-statement-alter-table.md)
```
Comment (hfxsd) [inline TOC.md:736]:
```suggestion
        - [Overview](/sql-statements/sql-statement-alter-table.md)
```

**Review by hfxsd — APPROVED** [review]

File: TOC-tidb-cloud.md [hunk 3]
```diff
@@ -360,21 +358,28 @@
...
-      - [`ALTER INDEX`](/sql-statements/sql-statement-alter-index.md)
       - [`ALTER INSTANCE`](/sql-statements/sql-statement-alter-instance.md)
       - [`ALTER PLACEMENT POLICY`](/sql-statements/sql-statement-alter-placement-policy.md)
       - [`ALTER RANGE`](/sql-statements/sql-statement-alter-range.md)
       - [`ALTER RESOURCE GROUP`](/sql-statements/sql-statement-alter-resource-group.md)
       - [`ALTER SEQUENCE`](/sql-statements/sql-statement-alter-sequence.md)
-      - [`ALTER TABLE`](/sql-statements/sql-statement-alter-table.md)
-      - [`ALTER TABLE COMPACT`](/sql-statements/sql-statement-alter-table-compact.md)
+      - `ALTER TABLE`
+        - [`Overview`](/sql-statements/sql-statement-alter-table.md)
```
Comment (Oreoxmt) [inline TOC-tidb-cloud.md:367]:
```suggestion
        - [Overview](/sql-statements/sql-statement-alter-table.md)
```

**Review by Oreoxmt — APPROVED** [review]
Rest LGTM

**Pull request merged**